### PR TITLE
feature/SP-167: Fixed tests and services

### PR DESCRIPTION
**Pull request by AliAkiltayev**
Previous SP-167 branch was not working properly, made these fixes for proper connection and auto resolution

### CODE REVIEW

File: client/src/main/java/io/mx51/spi/Spi.java [hunk 1]
```diff
@@ -2209,17 +2206,16 @@ private void autoResolveEftposAddress() throws IOException {
...
 
         DeviceService deviceService = new DeviceService();
-        Response httpResponse = deviceService.retrieveDeviceAddress(serialNumber, deviceApiKey, acquirerCode, inTestMode);
-        DeviceAddressStatus deviceAddressStatus = DeviceHelper.generateDeviceAddressStatus(httpResponse, eftposAddress);
+        DeviceAddressStatus deviceAddressStatus = deviceService.retrieveDeviceAddress(serialNumber, deviceApiKey, acquirerCode, inTestMode, eftposAddress);
         currentDeviceStatus = deviceAddressStatus;
 
-        if (deviceAddressStatus.getDeviceAddressResponseCode() != DeviceAddressResponseCode.DEVICE_SERVICE_ERROR) {
+        if (deviceAddressStatus.getDeviceAddressResponseCode() == DeviceAddressResponseCode.DEVICE_SERVICE_ERROR) {
             LOG.warn("Trying to auto resolve address, but device address has not changed.");
```
Comment (RaymondK-mx51) [inline client/src/main/java/io/mx51/spi/Spi.java:2213]: Can you change the log message to `"Could not communicate with device address service."`

Comment (AliAkiltayev) [reply client/src/main/java/io/mx51/spi/Spi.java]: Changed

File: client/src/main/java/io/mx51/spi/Spi.java [hunk 2]
```diff
@@ -2231,6 +2227,7 @@ private void autoResolveEftposAddress() throws IOException {
         conn.setAddress(eftposAddress);
 
         LOG.info("Address resolved to " + deviceAddressStatus.getAddress());
+        LOG.info("Address status#### " + deviceAddressStatus.getDeviceAddressResponseCode());
```
Comment (RaymondK-mx51) [inline client/src/main/java/io/mx51/spi/Spi.java:2230]: can you please remove this line? thanks

**Review by RaymondK-mx51 — CHANGES_REQUESTED** [review]
see comments

File: client/src/main/java/io/mx51/spi/Spi.java (hunk 2 above)
Comment (AliAkiltayev) [reply client/src/main/java/io/mx51/spi/Spi.java:2230]: Removed

**Comment by RaymondK-mx51** [comment]
@DixusSlim can you please review this PR. Thanks

**Pull request closed without merging**